sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1089,6 +1094,26 @@ def _path_residuals(X, y, train, test, path, path_params, alphas=None,
     y_train = y[train]
     X_test = X[test]
     y_test = y[test]
+    if sample_weight is None:
+        sw_train, sw_test = None, None
+    else:
+        sw_train = sample_weight[train]
+        sw_test = sample_weight[test]
+        n_samples = X_train.shape[0]
+        # simplify things by rescaling sw_train to sum up to n_samples on
+        # training set.
+        # => np.average(x, weights=sw) = np.mean(sw * x)
+        sw_train *= (n_samples / np.sum(sw_train))
+        # Objective function is:
+        # 1/2 * np.average(squared error, weights=sw) + alpha * penalty
+        # but coordinate descent minimizes:
+        # 1/2 * sum(squared error) + alpha * penalty
+        # enet_path therefore sets alpha = n_samples * alpha
+        # With sw, enet_path should set alpha = sum(sw) * alpha
+        # Therefore, we rescale alpha = sum(sw) / n_samples * alpha
+        # Note: As we rescaled sample_weight to sum up to n_samples,
+        #       we don't need this
+        # alpha *= np.sum(sample_weight) / n_samples
```

Comment:
I think this can be made easier to read. If I understood correctly,
I would propose the following:
```suggestion
        # Rescaling sw_train to sum up to n_samples on the training set.
        # In what follow:
        #      np.average(x, weights=sw) = np.mean(sw * x)
        #
        # The objective function here is
        #
        #     1/2 * np.average(squared error, weights=sw) + alpha * penalty
        #
        # but coordinate descent minimizes
        #
        #     1/2 * sum(squared error) + alpha * penalty
        #
        # Therefore enet_path sets
        #
        #     alpha = n_samples * alpha
        #
        #  With sw, enet_path should set
        #
        #     alpha = sum(sw) * alpha
        # 
        # Therefore, we rescale alpha = sum(sw) / n_samples * alpha
        # Note: As we previously have rescaled sample_weight to
        # sum up to n_samples, we don't need to perform
        # 
        #     alpha *= np.sum(sample_weight) / n_samples
        #
        # Hence the following value of sw_train
        sw_train *= (n_samples / np.sum(sw_train))
```

Reply:
Thanks, that reads better. I restructured this long comment based on your suggestion. As I think 2 times the same long comment is code bloat, I changed the original comment in ElasticNet.fit and referred to it.